AutoCollection/PreAggregatedMetrics.ts

```diff
@@ -0,0 +1,300 @@
+                    AutoCollectPreAggregatedMetrics._requestCountersCollection[i].lastTotalCount = AutoCollectPreAggregatedMetrics._requestCountersCollection[i].totalCount;
+                    AutoCollectPreAggregatedMetrics._requestCountersCollection[i].lastTime = +new Date;
+                }
+                for (let i = 0; i < AutoCollectPreAggregatedMetrics._exceptionCountersCollection.length; i++) {
+                    AutoCollectPreAggregatedMetrics._exceptionCountersCollection[i].lastTotalCount = AutoCollectPreAggregatedMetrics._exceptionCountersCollection[i].totalCount;
+                    AutoCollectPreAggregatedMetrics._exceptionCountersCollection[i].lastTime = +new Date;
+                }
+                for (let i = 0; i < AutoCollectPreAggregatedMetrics._traceCountersCollection.length; i++) {
+                    AutoCollectPreAggregatedMetrics._traceCountersCollection[i].lastTotalCount = AutoCollectPreAggregatedMetrics._traceCountersCollection[i].totalCount;
+                    AutoCollectPreAggregatedMetrics._traceCountersCollection[i].lastTime = +new Date;
+                }
+
+                this._collectionInterval = collectionInterval || this._collectionInterval;
+                this._handle = setInterval(() => AutoCollectPreAggregatedMetrics.trackPreAggregatedMetrics(), this._collectionInterval);
+                this._handle.unref(); // Allow the app to terminate even while this loop is going on
+            }
+        } else {
+            if (this._handle) {
+                clearInterval(this._handle);
+                this._handle = undefined;
+            }
+        }
+    }
+
+    public static countException(dimensions: MetricExceptionDimensions) {
+        if (!AutoCollectPreAggregatedMetrics.isEnabled()) {
+            return;
+        }
+        let counter: AggregatedMetricCounter = AutoCollectPreAggregatedMetrics._getAggregatedCounter(dimensions, this._exceptionCountersCollection);
+        counter.totalCount++;
+    }
+
+    public static countTrace(dimensions: MetricTraceDimensions) {
+        if (!AutoCollectPreAggregatedMetrics.isEnabled()) {
+            return;
+        }
+        let counter: AggregatedMetricCounter = AutoCollectPreAggregatedMetrics._getAggregatedCounter(dimensions, this._traceCountersCollection);
+        counter.totalCount++;
+    }
+
+    public static countRequest(duration: number | string, dimensions: MetricRequestDimensions) {
+        if (!AutoCollectPreAggregatedMetrics.isEnabled()) {
+            return;
+        }
+        let durationMs: number;
+        let counter: AggregatedMetricCounter = AutoCollectPreAggregatedMetrics._getAggregatedCounter(dimensions, this._requestCountersCollection);
+        if (typeof duration === 'string') {
+            // dependency duration is passed in as "00:00:00.123" by autocollectors
+            durationMs = +new Date('1970-01-01T' + duration + 'Z'); // convert to num ms, returns NaN if wrong
```

Comment:
Do (because caller is controlled by this code) or should we handle invalid values for strings?

Reply:
The string case is only for internal code, basically is receiving the duration string included in the generated envelope